Fortran/gfortran/regression/DisabledFiles.cmake

```diff
@@ -119,6 +119,22 @@ file(GLOB UNSUPPORTED_FILES CONFIGURE_DEPENDS
   unlimited_polymorphic_14.f90
   # Unsupported predefined macro: __TIMESTAMP__
   wdate-time.F90
+
+  # This test occasionally fails. It checks that two arrays initialized with
+  # random numbers are not identical. This might be the case in gfortran, but in
+  # flang, it is possible, though unlikely, for two "randomly initialized"
+  # arrays to be identical. Unless something in flang's implementation of the
+  # RANDOM_* intrinsics changes, this test will be unsupported.
```

Comment:
> It checks that two arrays initialized with random numbers are not identical.

While this description isn't wrong, it's slightly incomplete/inaccurate.

First, the test initializes two arrays from random real numbers (in the range 0-1), multiplied by 1e6, converted to integer and back to real. Some precision may be lost in this process, as only 6 digits are preserved.
Then the test checks if any element in the first array matches the corresponding element in the second array, aborting if true.
Because of the precision loss and random nature of the test, 2 elements may end up matching.

This is the first part of the test. The second part initializes the arrays again, but this time using a repeatable pseudo random number generator, and then checks if the arrays are equal, aborting if not. This part looks correct to me and can be omitted in the comment.

Maybe something like:
```
  # This test occasionally fails. It checks that two arrays initialized with
  # random real numbers, converted to integers, are not identical. This
  # might be the case in gfortran, but in flang, because of the precision
  # loss and random nature of the test, it is possible, though unlikely, for
  # two "randomly initialized" array elements to be identical. Unless
  # something in flang's implementation of the RANDOM_* intrinsics
  # changes, this test will be unsupported.
```

Although I guess this test is probably flaky with gfortran too.

Reply:
I wonder if flang's random number generation has some bias that is leading to producing the same numbers more often than gfortran. Even 6 digits should make it rare for these to fail.

But +1 for disabling it. This isn't a good test design because it will fail spuriously with small likelihood.

Reply:
I think flang may lack some randomness here, I opened https://github.com/llvm/llvm-project/issues/106221 with some more statistically relevant facts (the gfortran test should indeed do a bit of sampling, even gfortran will fail it once in 100000).

Reply:
Yes, I am not too keen on enabling this myself. We could consider adding a similar test of our own, but even then, I am not sure how to (or if we even should) test for "randomness".